app/boards/mimx93_evk_a55.overlay

```diff
@@ -58,4 +46,47 @@
 	host_ram: memory@80000000 {
 		reg = <0x80000000 DT_SIZE_M(1024)>;
 	};
+
+	sai3: dai@42660000 {
+		compatible = "nxp,dai-sai";
+		reg = <0x42660000 DT_SIZE_K(64)>;
+		mclk-is-output;
+		clocks = <&ccm IMX_CCM_SAI3_CLK 0x0 0x0>;
+		clock-names = "mclk1";
+		/* set to a period's worth of data (384B) so the SAI will
+		 * request a DMA transfer every 1ms (which is the period size)
+		 */
+		rx-fifo-watermark = <96>;
+		/* set to 1 because we want to avoid multiple DMA transfer requests
+		 * when the SAI is started. This way, the SAI will request data every
+		 * 1ms (which is the period size).
+		 */
+		tx-fifo-watermark = <1>;
+		/* can't let SOF use the actual FIFO depth (128 FIFO words) because
+		 * the DMA transfer size isn't divizible by it so the driver would
+		 * return an error. As such, set the FIFO depth to a period's worth
+		 * of data (384B).
+		 */
+		fifo-depth = <96>;
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_SPI 171 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+		dai-index = <3>;
+	};
+
+	edma4: dma@42000000 {
+		compatible = "nxp,edma";
+		reg = <0x42000000 (DT_SIZE_K(64) * 32)>;
+		valid-channels = <0>, <1>;
```

Comment:
If you have here `<0>, <1>` shouldn't be `dma-channels` based on this: https://github.com/zephyrproject-rtos/zephyr/pull/65671/commits/bba525577eb0e0528e1460cd2e43aa952958400d#diff-7e174a2aefc1b54220f6b815c04028a782f004bc4d5d03a39f604703f09b3234R16?

Reply:
> Do you mean having something like: `dma-channels = <2>` instead of `valid-channels = <0>, <1>`? 

Yes.

>If so, both approaches are correct. For a (**very**) high number of channels, the `dma-channels` "way" would be better since the lookup function would be O(1), but since this will never be the case I'm not sure choosing one approach over the other would make that much difference. If you believe the `dma-channels` approach is more intuitive then I don't mind switching to it.

Ok, let's keep` valid-channels`, is not much difference for 2 channels. 
We'll change the list (`<0>, <1>`) when we'll have support for multiple channels and the update will be easier having already the `valid-channels` option.